pkg/fileutil/preallocate_darwin.go

```diff
@@ -29,7 +29,13 @@ func preallocExtend(f *os.File, sizeInBytes int64) error {
 	return preallocExtendTrunc(f, sizeInBytes)
 }
 
+// blocks allocated for the file in 512-byte units
+// (see st_blocks in https://github.com/apple/darwin-xnu/blob/master/bsd/man/man2/stat.2)
+const blockSize int64 = 512
```

Comment:
we should be able to get blocksize programmatically. osx also supports 4k blocksize.